File: examples/php/Employee.php
<?php
interface IEmployee {
    public function getName();
    public function getSalary();
    public function displayEmployee();
}

class Employee implements IEmployee {
    var $name;
    var $salary;

    public function getName() {
        return $this->name;
    }

    protected function setName($name) {
        $this->name = (string)$name;
    }

    public function getSalary() {
        return $this->salary;
    }

    protected function setSalary($salary) {
        $this->salary = floatval($salary);
    }

    public function __construct() {
        $numargs = func_num_args();
        if ($numargs < 2 || $numargs > 2) {
            $this->setName('');
            $this->setSalary(0);
        } else /* Overload */ {
            $this->setName(func_get_arg(0));
            $this->setSalary(func_get_arg(1));
        }
    }

    public function __toString() {
        return sprintf("Name: %s, Salary: $%.2f", $this->getName(), $this->getSalary());
    }

    public function displayEmployee() {
        print(nl2br($this.PHP_EOL));
    }
}

function main() {
    $emp1 = new Employee("Bob", 75000);
    $emp2 = new Employee("Joe", 50000);
    $emp1->displayEmployee();
    $emp2->displayEmployee();
}

main();
?>

File: examples/java/Employee.java
interface Employee {
	public String getName();

	public float getSalary();

	public void displayEmployee();
}

class EmployeeImpl extends Object implements Employee {
	private String name;
	private float salary;

	@Override
	public String getName() {
		return this.name;
	}

	protected void setName(String name) {
		this.name = name;
	}

	@Override
	public float getSalary() {
		return this.salary;
	}

	protected void setSalary(float salary) {
		this.salary = salary;
	}

	public EmployeeImpl(String name, float salary) {
		setName(name);
		setSalary(salary);
	}

	@Override
	public String toString() {
		return String.format("Name: %s, Salary: $%.2f", getName(), getSalary());
	}

	@Override
	public void displayEmployee() {
		System.out.println(this);
	}
}

class Main {
	public static void main(String[] args) {
		Employee emp1 = new EmployeeImpl("Bob", 75000);
		Employee emp2 = new EmployeeImpl("Joe", 50000);
		emp1.displayEmployee();
		emp2.displayEmployee();
		System.exit(0);
	}
}